src/drivers/config/drivers.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) Microsoft Corporation
+# All rights reserved.
+#
+# MIT License
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+
+import logging
+import logging.config
+
+
+class Drivers:
+
+    def __init__(self, cluster_configuration, service_configuration, default_service_configuraiton):
+        self.logger = logging.getLogger(__name__)
+
+        self.cluster_configuration = cluster_configuration
+        self.service_configuration = self.merge_service_configuration(service_configuration, default_service_configuraiton)
+
+
+
+    def merge_service_configuration(self, overwrite_srv_cfg, default_srv_cfg):
```

Comment:
Hi, Yundong, is it better to write a superclass implementing `run` and `merge_service_configuration`, interfacing `validation_pre` and `validation_post`? Then every service config model inherit this superclass and implement the interface. I think this merge function works for most cases, no need to rewrite it in every service.

If some service has customization requirements, it could overwrite `run` function.

Reply:
agreed

Reply:
@mzmssg Can you send pr to implement it?

Reply:
@ydye Get you, will have a try